src/dbnode/namespace/dynamic_test.go

```diff
@@ -325,6 +300,97 @@ func TestInitializerUpdateSuccess(t *testing.T) {
 	require.NoError(t, reg.Close())
 }
 
+func TestInitializerAllowEmptyEnabled_EmptyRegistry(t *testing.T) {
+	defer leaktest.CheckTimeout(t, time.Second)()
+
+	ctrl := gomock.NewController(t)
```

Comment:
nit: use `xtest.NewController` instead, it's a bit better about failing on unexpected calls

Reply:
lol, I know why I used `gomock` here. Unfortunately, we've introduced a circular dependency at some point so now `xtest` depends on `dbnode/namespace`. I've got a todo to look into it, but I think it has something to do with the extended options changes.

Reply:
Ah lol, yeah I had one of those too, but mine was a lot simpler to detangle :p

Maybe at some point we just pull out `xtest.Controller` to its own `xgomock.Controller` package or something?